File: java/OauthTest2/src/main/java/com/adrianpoplesanu/OauthTest2/controller/HelloController.java
package com.adrianpoplesanu.OauthTest2.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import static org.springframework.http.HttpStatus.OK;

@RestController
@RequestMapping("/api/v1/hello")
@RequiredArgsConstructor
@Validated
public class HelloController {
    @GetMapping(produces = MediaType.TEXT_PLAIN_VALUE)
    @ResponseStatus(OK)
    public ResponseEntity<String> hello() {
        return ResponseEntity.ok("Hello Second Version");
    }
}

File: cursor-generated/uefa1/matches/src/main/kotlin/ro/adrianus/matches/controller/HelloController.kt
package ro.adrianus.matches.controller

import org.slf4j.LoggerFactory
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import ro.adrianus.matches.service.FootballDataService

@RestController
class HelloController(
    private val footballDataService: FootballDataService
) {

    private val log = LoggerFactory.getLogger(javaClass)

    @GetMapping("/hello")
    fun hello(): ResponseEntity<String> {
        log.info("GET /hello requested")
        return ResponseEntity.ok("Hello, World!")
    }

    @GetMapping("/matches")
    fun matches(): ResponseEntity<String> {
        log.info("GET /matches requested")
        val result = footballDataService.retrieveMatches()
        return ResponseEntity.ok(result ?: "No matches found")
    }
}
